src/router/index.js

```diff
@@ -34,17 +35,24 @@ function createRouter() {
       {path:'/login', name: 'login', component:LoginView, meta: {noAuth: true}},
       {path:'/recordings',component:RecordingsView},
       {path:'/register',component:RegisterView, meta: {noAuth: true}},
-      {path:'/video/:id',component:VideoView}
+      {path:'/video/:id',component:VideoView},
+      {path:'/add_email', name: 'addEmail', component: AddEmailView, meta: {noEmail: true}},
     ]
   });
 
   router.beforeEach((to, from, next) => {
     const isLoggedIn = store.getters['User/isLoggedIn'];
-
-    if (isLoggedIn) {
-      if(to.name === 'login' || to.name === 'register') {
+    const hasEmail = store.getters['User/hasEmail'];
+    if (isLoggedIn && hasEmail) {
+      if(to.name === 'login' || to.name === 'register' || to.name === 'addEmail') {
```

Comment:
Not a biggy but might be better to do something like:

if(['login','register','addEmail'].includes(to.name)) {
...